plugin/src/main/java/com/iridium/iridiumcore/utils/ItemStackUtils.java

```diff
@@ -81,8 +79,18 @@ public static ItemStack makeItem(Item item) {
             if (item.material == XMaterial.PLAYER_HEAD && item.headData != null) {
                 return setHeadData(item.headData, itemstack);
             } else if (item.material == XMaterial.PLAYER_HEAD && item.headOwner != null) {
+                UUID uuid;
+                //OfflinePlayer will return a random uuid if the player doesn't exist in server cache
                 OfflinePlayer offlinePlayer = Bukkit.getServer().getOfflinePlayer(item.headOwner);
-                return setHeadData(SkinUtils.getHeadData(offlinePlayer.getUniqueId()), itemstack);
+                if (offlinePlayer.hasPlayedBefore()) {
+                    uuid = offlinePlayer.getUniqueId();
+                } else {
+                    if ((uuid = SkinUtils.getUUID(item.headOwner)) == null) {
```

Comment:
Especially this line here, im too dumb to understand it

Reply:
This line sets the uuid to SkinUtils.getUUID(item.headOwner) and checks if it's null simultaneously